shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -110,12 +111,37 @@ public void platformPlugin_hasStrings() {
     PlatformChannel fakePlatformChannel = mock(PlatformChannel.class);
     PlatformPlugin platformPlugin = new PlatformPlugin(fakeActivity, fakePlatformChannel);
 
+    // Plain text
     ClipData clip = ClipData.newPlainText("label", "Text");
     clipboardManager.setPrimaryClip(clip);
     assertTrue(platformPlugin.mPlatformMessageHandler.clipboardHasStrings());
 
+    // Empty plain text
     clip = ClipData.newPlainText("", "");
     clipboardManager.setPrimaryClip(clip);
+    // Without actually accessing clipboard data (preferred behavior), it is not possible to
+    // distinguish between empty and non-empty string contents.
+    assertTrue(platformPlugin.mPlatformMessageHandler.clipboardHasStrings());
+
+    // HTML text
+    clip = ClipData.newHtmlText("motto", "Don't be evil", "<b>Don't</b> be evil");
+    clipboardManager.setPrimaryClip(clip);
+    assertTrue(platformPlugin.mPlatformMessageHandler.clipboardHasStrings());
+
+    // Text MIME type
+    clip = new ClipData("label", new String[] {"text/something"}, new ClipData.Item("content"));
+    clipboardManager.setPrimaryClip(clip);
+    assertTrue(platformPlugin.mPlatformMessageHandler.clipboardHasStrings());
+
+    // Other MIME type
+    clip =
+        new ClipData(
+            "label", new String[] {"application/octet-stream"}, new ClipData.Item("content"));
+    clipboardManager.setPrimaryClip(clip);
+    assertFalse(platformPlugin.mPlatformMessageHandler.clipboardHasStrings());
+
+    // Empty clipboard
+    clipboardManager.clearPrimaryClip();
     assertFalse(platformPlugin.mPlatformMessageHandler.clipboardHasStrings());
```

Comment:
can it also verify that API to get the text isn't called? https://github.com/flutter/engine/pull/27215/files#diff-8dfd4b72efbe5f5b494de945a284feba1d7e63a71e56a567235e6748717801caR465